tests/pool.rs

```diff
@@ -67,28 +67,25 @@ fn configure_template_update_interval() {
 async fn basic_initialization_flow() {
     let pool = TestPool::spawn_with_args("--start-diff 0.00001");
 
-    let mut client = pool.stratum_client().await;
+    let client = pool.stratum_client().await;
+    let mut events = client.connect().await.unwrap();
 
-    let (subscribe, _, _) = client.subscribe(USER_AGENT.into()).await.unwrap();
+    let (subscribe, _, _) = client.subscribe().await.unwrap();
 
     assert_eq!(subscribe.subscriptions.len(), 2);
 
     assert!(client.authorize().await.is_ok());
 
-    let set_difficulty = match client.incoming.recv().await.unwrap() {
-        Message::Notification { method: _, params } => {
-            serde_json::from_value::<SetDifficulty>(params).unwrap()
-        }
-        _ => panic!(),
+    let difficulty = match events.recv().await.unwrap() {
+        stratum::Event::SetDifficulty(difficulty) => difficulty,
+        _ => panic!("Expected SetDifficulty"),
     };
 
-    assert!(set_difficulty.difficulty() == Difficulty::from(0.00001));
+    assert!(difficulty == Difficulty::from(0.00001));
```

Comment:
```suggestion
    assert_eq!(difficulty, Difficulty::from(0.00001));
```

Equality check in assert, use `assert_eq`.